packages/async-flow/src/replay-membrane.js

```diff
@@ -0,0 +1,389 @@
+            ` with non-empty callstack ${q(callStack)};`;
+      Fail`Called from a previous run: ${guestTarget}${b(extraDiagnostic)}`;
+    }
+    /** @type {Outcome} */
+    let outcome;
+    try {
+      const guestEntry = harden([
+        'checkCall',
+        guestTarget,
+        optVerb,
+        guestArgs,
+        callIndex,
+      ]);
+      if (log.isReplaying()) {
+        const entry = log.nextEntry();
+        equate(
+          guestEntry,
+          entry,
+          `replay ${callIndex}:
+     ${q(guestEntry)}
+  vs ${q(entry)}
+    `,
+        );
+        outcome = /** @type {Outcome} */ (nestInterpreter(callIndex));
+      } else {
+        const entry = guestToHost(guestEntry);
+        log.pushEntry(entry);
+        const [_, ...args] = entry;
+        nestInterpreter(callIndex);
+        outcome = performCall(...args);
+      }
+    } catch (fatalError) {
+      throw panic(fatalError);
+    }
+
+    if (outcome.kind === 'return') {
+      return outcome.result;
+    } else {
+      outcome.kind === 'throw' ||
+        // @ts-expect-error TS correctly knows this case would be outside
+        // the type. But that's what we want to check.
+        Fail`unexpected outcome kind ${q(outcome.kind)}`;
+      throw outcome.problem;
+    }
+  };
+
+  // //////////////// Converters ///////////////////////////////////////////////
+
+  const makeGuestForHostRemotable = hRem => {
+    let gRem;
+    /** @param {PropertyKey} [optVerb] */
+    const makeGuestMethod = (optVerb = undefined) => {
+      const guestMethod = (...guestArgs) => {
+        const callIndex = log.getIndex();
+        return guestCallsHost(gRem, optVerb, guestArgs, callIndex);
+      };
+      if (optVerb) {
+        defineProperties(guestMethod, {
+          name: { value: String(optVerb) },
+          length: { value: Number(hRem[optVerb].length || 0) },
+        });
+      } else {
+        defineProperties(guestMethod, {
+          name: { value: String(hRem.name || 'anon') },
+          length: { value: Number(hRem.length || 0) },
+        });
+      }
+      return guestMethod;
+    };
+    const iface = String(getInterfaceOf(hRem) || 'remotable');
+    const guestIface = `${iface} guest wrapper`; // just for debugging clarity
+    if (typeof hRem === 'function') {
+      // NOTE: Assumes that a far function has no "static" methods. This
+      // is the current marshal design, but revisit this if we change our
+      // minds.
+      gRem = Remotable(guestIface, undefined, makeGuestMethod());
+      // NOTE: If we ever do support that, probably all we need
+      // to do is remove the following `throw Fail` line.
+      throw Fail`host far functions not yet passable`;
+    } else {
+      const methodNames = getMethodNames(hRem);
+      const guestMethods = methodNames.map(name => [
+        name,
+        makeGuestMethod(name),
+      ]);
+      // TODO in order to support E *well*,
+      // use HandledPromise to make gRem a remote presence for hRem
```

Comment:
The problem with not implementing E right away is that any usage of eventual-send on a remote presence will currently result in an exception thrown, and when we update the asyncFlow helper, it will become a success, with an observable log difference, so any in-progress flow invocation will fail to replay unless extra steps are taken.

Reply:
Agreed that we cannot commit to any log state until this is fixed. We're about to work on this, but I think I still want to do it in a separate PR staged on this one. So, resolving.